### PR TITLE
add support for KPI and analytical dashboard metadata

### LINKED ISSUE #797: Add support for KPI and analytical dashboard metadata
We should be able to: 
- get, query, create or update analytical dashboard
- get, query, create or update KPI on analytical dashboard
- get, query, create or update KPI alert

### PR DESCRIPTION
**Pull request by peter-plochan**
Fixes #797

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-java-model/src/test/groovy/com/gooddata/sdk/model/md/dashboard/filter/DashboardDateFilterTest.groovy [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    @Unroll
+    def "creates new #type filter and serializes to JSON"() {
+        expect:
+        that filter, jsonEquals(readStringFromResource(filterJson(type)))
+
+        where:
+        type     | filter
+        RELATIVE | relativeDateFilter(-6, -1, GRANULARITY, DATE_DATASET_URI)
+        ABSOLUTE | absoulteDateFilter(LocalDate.of(2019, 06, 01), LocalDate.of(2019, 06, 30), DATE_DATASET_URI)
```
Comment (jimirocks) [inline gooddata-java-model/src/test/groovy/com/gooddata/sdk/model/md/dashboard/filter/DashboardDateFilterTest.groovy]: typo: `absoulteDateFilter`

File: gooddata-java-model/src/main/java/com/gooddata/sdk/model/md/dashboard/AnalyticalDashboard.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
+@JsonTypeName("analyticalDashboard")
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class AnalyticalDashboard extends AbstractObj implements Queryable, Updatable {
+
+    private static final long serialVersionUID = -2217112261771452747L;
+
+    private final Content content;
+
+    public AnalyticalDashboard(final String title, final List<String> widgetUris, final String filtersContextUri) {
```
Comment (jimirocks) [inline gooddata-java-model/src/main/java/com/gooddata/sdk/model/md/dashboard/AnalyticalDashboard.java]: add javadoc

File: gooddata-java-model/src/main/java/com/gooddata/sdk/model/md/dashboard/AnalyticalDashboard.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        this.content = content;
+    }
+
+    @JsonProperty
+    private Content getContent() {
+        return content;
+    }
+
+    @JsonIgnore
+    public List<String> getWidgetUris() {
```
Comment (jimirocks) [inline gooddata-java-model/src/main/java/com/gooddata/sdk/model/md/dashboard/AnalyticalDashboard.java]: add javadoc to all public getters

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-java-model/src/test/groovy/com/gooddata/sdk/model/md/dashboard/filter/DashboardDateFilterTest.groovy (hunk 1 above)
Comment (peter-plochan) [reply gooddata-java-model/src/test/groovy/com/gooddata/sdk/model/md/dashboard/filter/DashboardDateFilterTest.groovy]: fixed

File: gooddata-java-model/src/main/java/com/gooddata/sdk/model/md/dashboard/AnalyticalDashboard.java (hunk 2 above)
Comment (peter-plochan) [reply gooddata-java-model/src/main/java/com/gooddata/sdk/model/md/dashboard/AnalyticalDashboard.java]: fixed

File: gooddata-java-model/src/main/java/com/gooddata/sdk/model/md/dashboard/AnalyticalDashboard.java (hunk 3 above)
Comment (peter-plochan) [reply gooddata-java-model/src/main/java/com/gooddata/sdk/model/md/dashboard/AnalyticalDashboard.java]: fixed

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-java-merge-compliance-check/105/ : SUCCESS in 16s

**Pull request merged**